### DISCUSSION TITLE
Temperature Dependent Anisotrpic Heat Transfer

### DISCUSSION THREAD

**Discussion by bhollra** [idaholab/moose#18082] [Q&A Modules: General] [answered]
Greetings all,

I would like to implement a model for ansiotropic heat transfer with temperature dependent thermal conductivities. I have looked at the [AnisoHeatConduction](https://mooseframework.inl.gov/source/kernels/AnisoHeatConduction.html) kernel with [AnisoHeatConductionMaterial](https://mooseframework.inl.gov/source/materials/AnisoHeatConductionMaterial.html), but it was not clear to me how temperature dependency can be included. The only option I see to create a variable thermal conductivity is with the _thermal_conductivity_x_pp_ parameter, but this accepts only a scalar post processor, so it seems a spatially varying material property is not possible.  

I did not see any other Kernel or Material blocks that looked capable of solving this problem. Is temperature dependent, anisotropic heat transfer possible in MOOSE? 

Thank you!

**Comment by singhgp4321** [comment, marked as answer]
#17783 might be what you are planning to use. This pull request is under review and should get merged soon . . . in about a week or so I think.

**Reply by bhollra** [reply]
Thanks for the heads up, looking forward to using the new version!

**Reply by hamedattariani** [reply]
> #17783 might be what you are planning to use. This pull request is under review and should get merged soon . . . in about a week or so I think.

Does this kernel work for user-defined temperature-dependent thermal conductivity and specific heat? I tried to define them as ParsedMaterial, but I am getting errors.

**Reply by GiudGiud** [reply]
It should but you need to specify anisotropic material properties, so typically vectors or tensors-values, not Real-valued.
For this new kernel it seems you need to use a RankTwo tensor

How are you defining the `ParsedMaterial` ?
You can try with something like this
https://mooseframework.inl.gov/source/materials/ADGenericFunctionRankTwoTensor.html

**Reply by hamedattariani** [reply]
I used "DerivativeParsedMaterial" and also "ParsedMaterial" in the input script for  "Thermal Conductivity" and "specific heat". I got errors for both scenarios. Error" Unable to find function "specific heat"

The ADGenericFunctionRankTwoTensor is indeed interesting. But, the issue is the ParsedFunction can be an "expression as a function of time (t) or coordinate (x, y, or z)". I need it as a function of the temperature. Please, correct me if I am wrong and making a fundamental mistake here. 

Example:
I defined KTX, etc as  the "DerivativeParsedMaterial" or "ParsedMaterial" in the material block 
**[Materials]**
[heat1]
    type = AnisoHeatConductionMaterial
    temperature = temperature
    specific_heat = **specific_heat**
    thermal_conductivity = 'K_TX K_TY K_TZ 0 0 0 0 0 0'
  []

[./**specific_heat** ]  
    type = ParsedMaterial
    f_name = specific_heat 
    args = 'temperature'
     function = '(1/(1+exp(-2**(temperature-273))))*(483.04+0.215*temperature)' 
    outputs = exodus
[../]

**Reply by GiudGiud** [reply]
So this field should be a function not a material property. That s what the error is complaining about

you ll need to rework that material a bit if you want a temperature dependence.
Alternatively you could try to use another material that supports it, maybe splitting up the three properties in three materials.

**Reply by singhgp4321** [reply]
@hamedattariani Checkout out #22028 and the included test. Seems like that is what you want? If yes, I will put it on my priority list to get it merged.

**Reply by hamedattariani** [reply]
@singhgp4321 Yessss, thanks, that is exactly what I need.

**Reply by singhgp4321** [reply]
Okay, I'll work on it and let you know when it is ready . . . may be 1-2 weeks. If you want it earlier than that then may be you can just copy the code from the PR.

**Reply by hamedattariani** [reply]
@singhgp4321 Thanks! I just copied the files from PR and compiled the code. It works well.

**Reply by singhgp4321** [reply]
Excellent! @hamedattariani
